File: inventory_frontend/src/types/order.ts
export interface OrderItemPayload {
    product_id: number;
    quantity: number;
}

export interface OrderCreatePayload {
    customer_id: number;
    items: OrderItemPayload[];
}

export interface Order {
    id: number;
    customer_id: number;
    total_amount: number;
}

export interface OrderItem {
    product_id: number;
    quantity: number;
    price: number;
    subtotal: number;
}

export interface OrderDetail {
    id: number;
    customer_id: number;
    total_amount: number;
    items: OrderItem[];
}

File: inventory_backend/app/schemas/order.py
from decimal import Decimal

from pydantic import BaseModel, ConfigDict


class OrderItemCreate(BaseModel):
    product_id: int
    quantity: int


class OrderCreate(BaseModel):
    customer_id: int
    items: list[OrderItemCreate]


class OrderResponse(BaseModel):
    id: int
    customer_id: int
    total_amount: Decimal

    model_config = ConfigDict(
        from_attributes=True
    )


class OrderItemResponse(BaseModel):
    product_id: int
    quantity: int
    price: Decimal
    subtotal: Decimal

    model_config = ConfigDict(
        from_attributes=True
    )


class OrderDetailResponse(BaseModel):
    id: int
    customer_id: int
    total_amount: Decimal
    items: list[OrderItemResponse]